File: distributions/3D_smooth1_input.py
import numpy as np
from scipy.stats import beta as beta_dist

prefix = "3D_smooth1"
description = "Smooth, near-symmetric, mildly-concentrated independent Beta product (baseline case)"
dim = 3

D = 3
SHAPES = [(4.0, 4.0), (5.0, 5.0), (4.5, 4.5)]


def extract_samples(size):
    rng = np.random.default_rng()
    samples = np.zeros((size, D))
    for j, (a, b) in enumerate(SHAPES):
        samples[:, j] = rng.beta(a, b, size=size)
    return samples


def compute_pdf(points):
    points = np.asarray(points)
    density = np.ones(len(points))
    for j, (a, b) in enumerate(SHAPES):
        density *= beta_dist.pdf(points[:, j], a, b)
    return density

File: distributions/3D_smooth1_input.R
# 3D_smooth1: smooth, near-symmetric, mildly-concentrated independent Beta
# product -- the "easy"/baseline case for median vs midpoint splitting.
#
# Rationale: this scenario has NO sharp features, NO strong off-center skew,
# and NO structure that midpoint splitting would handle poorly -- each
# marginal is a mild, roughly-symmetric Beta centered near 0.5, matching
# midpoint's fixed geometric split almost exactly. Median and midpoint
# splitting are expected to perform SIMILARLY here; this is the baseline
# against which the other two (sharper, more asymmetric) scenarios should
# show a growing median advantage.

d <- 3
shapes <- list(c(4, 4), c(5, 5), c(4.5, 4.5))

extract_samples <- function(n) {
  samples <- matrix(0, nrow = n, ncol = d)
  for (j in 1:d) {
    samples[, j] <- rbeta(n, shapes[[j]][1], shapes[[j]][2])
  }
  return(samples)
}

compute_pdf <- function(points) {
  density <- rep(1, nrow(points))
  for (j in 1:d) {
    density <- density * dbeta(points[, j], shapes[[j]][1], shapes[[j]][2])
  }
  return(density)
}
